src/main/fc/tasks.c

```diff
@@ -160,21 +160,51 @@ static void taskUpdateAccelerometer(timeUs_t currentTimeUs)
 
 static void taskUpdateRxMain(timeUs_t currentTimeUs)
 {
-    if (!processRx(currentTimeUs)) {
-        return;
-    }
+    static enum {
+        CHECK, PROCESS, MODES, UPDATE
+    } state = CHECK;
+
+    switch (state) {
+    default:
+    case CHECK:
+        ignoreTaskTime();
+        rxProcessPending(true);
```

Comment:
To me, using `rxProcessPending()` to subvert the function of the task checker functions for the purpose of having the state machine executed even if there is nothing to be processed is doing it the wrong way round - we are building a shim to work around how our own task management works.
I think a better way will be to 'pull down' the state check of the state machine into the task checker function.

Reply:
`rxProcessPending()` isn't subverting the function really; it's basically saying "OK, so I know I needed to start a pass through the state machine, and I'm not done yet, so please let me finish what I started". I can have the checker check the state machine state instead and always return true if the state isn't `CHECK`. Please latest push for an implementation of this.